extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.smallrye.context.runtime;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.eclipse.microprofile.context.ThreadContext;
+
+import io.smallrye.context.impl.ManagedExecutorImpl;
+import io.smallrye.context.impl.ThreadContextImpl;
+
+@ApplicationScoped
+public class SmallRyeContextPropagationProvider {
+
+    private volatile ManagedExecutorImpl managedExecutor;
+
+    void initialize(ExecutorService executorService) {
+        managedExecutor = new ManagedExecutorImpl(-1, -1, (ThreadContextImpl) getAllThreadContext(), executorService, "no-ip") {
+            @Override
+            public void shutdown() {
+                throw new IllegalStateException("This executor is managed by the container and cannot be shut down.");
+            }
+
+            @Override
+            public List<Runnable> shutdownNow() {
+                throw new IllegalStateException("This executor is managed by the container and cannot be shut down.");
+            }
+        };
+    }
+
+    @Produces
+    @Singleton
+    public ThreadContext getAllThreadContext() {
```

Comment:
I'd rather not have these as they will instantly clash with addition of injection support in https://github.com/quarkusio/quarkus/issues/2243. We should pick one of the other approaches discussed on issue - either SR-specific API (but that kills the spec purpose somewhat) or make it so that users can only spawn executors from Quarkus thread pool (which would allow users to stick to spec API only). I am not sure if it even makes sense to let users create executors from both sources?
Also, with current design, you differentiate the pools used (Quarkus vs new) by using builder or injection. That doesn't seem very user friendly to me.

Reply:
Well no they won't because we don't have #2243 yet and when we do I can remove those. As far as users are concerned their code will keep working the same.